client/app/validators/validate-number-if.js

```diff
@@ -0,0 +1,19 @@
+import {
+  validateNumber,
+} from 'ember-changeset-validations/validators';
+
+export default function validateNumberIf(options) {
+  const { withoutValue, on } = options;
+
+  return (...args) => {
+    const [,,, changes, content] = args;
+    const target = Object.prototype.hasOwnProperty.call(changes, on) ? changes[on] : content[on];
+    const hasMatchingWith = target !== withoutValue;
+
+    if (hasMatchingWith) {
+      return validateNumber(options)(...args);
+    }
+
+    return true;
+  };
+}
```

Comment:
Ah, cool! I wonder if there's a way we could "curry" some generic "hasMatchingWith" function with a given validator:

```javascript
conditionallyValidate(validateNumber, {
  // ...normal options
})
```